### DISCUSSION TITLE
issue: bad setState() call inside `Controller` error when early returning and accessing `formState.isValid`

### DISCUSSION THREAD

**Discussion by rczobor** [react-hook-form/react-hook-form#11760] [Q&A]
### Version Number

7.51.2

### Codesandbox/Expo snack

https://codesandbox.io/p/devbox/vigilant-ully-rk9f55?file=%2Fsrc%2FApp.tsx%3A55%2C1&workspaceId=66cb3c57-3d8e-443f-853d-933dd2794011

### Steps to reproduce

1. Open console
2. Reload page
3. Error appears in console

Error is gone if:
1. Remove line where`form.formState.isValid` is logged 
2. Remove loading state early return
3. Remove either input controller
4. Downgrade to 7.51.0

### Expected behaviour

No errors in the console

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
Warning: Cannot update a component (`App`) while rendering a different component (`Controller`). To locate the bad setState() call inside `Controller`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
    at Controller (https://rk9f55-5173.csb.app/node_modules/.vite/deps/chunk-H6KY5HVK.js?v=bebf4d94:309:35)
    at FormField (https://rk9f55-5173.csb.app/src/components/ui/form.tsx:32:6)
    at form
    at FormProvider (https://rk9f55-5173.csb.app/node_modules/.vite/deps/chunk-H6KY5HVK.js?v=bebf4d94:80:11)
    at div
    at App (https://rk9f55-5173.csb.app/src/App.tsx?t=1712260786511:43:16)
    at QueryClientProvider (https://rk9f55-5173.csb.app/node_modules/.vite/deps/@tanstack_react-query.js?v=93d0a616:2635:3)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tomsnep** [comment]
I'm also experiencing the same behaviour since `7.51.1`. Should  dependencies on `formState` be treated differently? IMO this sounds like a breaking change/bug? @bluebill1049

**Reply by omridevk** [reply]
same here

**Reply by Michael-AT-Corporation** [reply]
same

**Reply by tgdn** [reply]
Same here

**Reply by abe1272001** [reply]
The same.

**Reply by antonchuvirau** [reply]
Wasted almost a day trying to find why that was happening in my app. Hopefully they will be a fix soon

**Reply by sim391calado** [reply]
I just noticed the same problem

**Reply by gastonyte** [reply]
Same problem here, why was this issue closed when apparently it is not solved? 
I can confirm that this react warning starts with version 7.51.1

**Reply by AnaisGalisson-ChillStudio** [reply]
still there : 7.51.4, why whas the initial issue moved as discussion ?

**Reply by robcurtis** [reply]
same here

**Reply by erwin-gapai** [reply]
I noticed this will happen if we dont provide defaultValue for each field.
instead of `undefined`, set the value to `''` or `null` for the defaultValue.

**Reply by kasparkallas** [reply]
Same here...

Used this work-around:
```ts
const { isValid, isValidating } = useFormState({ control });

const [isButtonDisabled, setIsButtonDisabled] = useState(true);
useEffect(() => {
  setIsButtonDisabled(!isValid || isValidating);
}, [isValid, isValidating]);
```

EDIT: Actually I'm not sure this solves much, but I think it guarantees another re-render which rectifies the first unfinished render.
